src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -2921,7 +2921,8 @@ namespace WaterThermalTanks {
 					WaterThermalTank( WaterThermalTankNum ).HeaterHeight1 = rNumericArgs( 7 );
 
 					//Test if Heater height is within range
-					if ( ( !WaterThermalTank( WaterThermalTankNum ).HeightWasAutoSized ) && ( WaterThermalTank( WaterThermalTankNum ).HeaterHeight1 > WaterThermalTank( WaterThermalTankNum ).Height ) ) {
+					Real64 tankHeightForTesting = 2.0 * sqrt( ( WaterThermalTank( WaterThermalTankNum ).Volume / WaterThermalTank( WaterThermalTankNum ).Height ) / DataGlobals::Pi );
```

Comment:
@Myoldmopar @JasonGlazer I think tankHeightForTesting needs to be sensitive to the Tank Shape type.